civismlext/preprocessing.py

```diff
@@ -174,6 +174,15 @@ def _expand_col(self, X, col):
         # then it's a category not seen during the fit and should
         # be ignored in the expansion.
         catcol = X[col].astype('category')
+
+        # check for overlap between categories in col and original fit levels
+        overlap = len([lvl for lvl in catcol.cat.categories if
+                       lvl in self.levels_[col]])
+        if overlap == 0:
+            warn_msg = "No overlap between levels in column " + \
+                       "'%s' and levels seen during fit" % (col)
+            warnings.warn(warn_msg, UserWarning)
```

Comment:
Why `UserWarning`? I see that this class is already issuing two kinds of warning; one is a `UserWarning` and the other is a `RuntimeWarning`. Is there a reason to use one rather than another?

Reply:
I don't have a strong preference, but I picked `UserWarning` since the issue is probably caused by something in the user input. It's also the default warning type, for what that's worth. Do you think `RuntimeWarning` makes more sense?

Reply:
I don't have a preference, I was just curious why you picked `UserWarning`. That logic makes sense.